VisitorHelper Addition

UpFault opened the pull request:
onMouseClicked in VisitorHelper.java now copies the item amount to clipboard

kevinthegreat1 commented:
Would be nice to have a text saying `[Click to Copy Amount]` or `[Copy Amount]`

UpFault commented:
rather than just copying it, yeah I can definitely make that happen.

kevinthegreat1 commented on src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java:
```diff
@@ -69,19 +69,25 @@ public static void onMouseClicked(double mouseX, double mouseY, int mouseButton,
+                if (isMouseOverText(mouseX, mouseY, copyAmountTextX, yPosition, textRenderer.getWidth("[Copy Amount]"), textHeight)) {
                     MinecraftClient.getInstance().keyboard.setClipboard(String.valueOf(itemEntry.getIntValue()));
                     return;
                 }
```
Change the x to not overlap and add some feedback to the player. You will have to import some stuff.
```suggestion
                MinecraftClient client = MinecraftClient.getInstance();
                if (client.player != null && isMouseOverText(mouseX, mouseY, TEXT_START_X + TEXT_INDENT + textWidth, yPosition, textRenderer.getWidth(" [Copy Amount]"), textHeight)) {
                    client.keyboard.setClipboard(String.valueOf(itemEntry.getIntValue()));
                    client.player.sendMessage(Constants.PREFIX.get().append("Copied amount successfully"));
                    return;
                }
```

UpFault replied:
gotta fix the clipboard, doesn't wanna copy with this change :/

kevinthegreat1 commented on src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java:
```diff
@@ -135,7 +141,9 @@ private static void drawScreenItems(DrawContext context, TextRenderer textRender
         int index = 0;
         for (Map.Entry<Pair<String, String>, Object2IntMap<String>> visitorEntry : itemMap.entrySet()) {
             Pair<String, String> visitorName = visitorEntry.getKey();
+            Text visitorNameText = Text.literal(visitorName.left());
             drawTextWithOptionalUnderline(context, textRenderer, Text.literal(visitorName.left()), TEXT_START_X, TEXT_START_Y + index * (LINE_SPACING + textRenderer.fontHeight), mouseX, mouseY);
+            drawTextWithOptionalUnderline(context, textRenderer, Text.literal("[Copy Amount]"), textRenderer.getWidth(visitorNameText) + 7, TEXT_START_Y + index * (LINE_SPACING + textRenderer.fontHeight), mouseX, mouseY);
```
```suggestion
            drawTextWithOptionalUnderline(context, textRenderer, Text.literal(visitorName.left() + " [Copy Amount]"), TEXT_START_X, TEXT_START_Y + index * (LINE_SPACING + textRenderer.fontHeight), mouseX, mouseY);
```

kevinthegreat1 commented on src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java:
```diff
@@ -69,19 +69,25 @@ public static void onMouseClicked(double mouseX, double mouseY, int mouseButton,
                 yPosition += LINE_SPACING + textHeight;
             }
         }
     }
```
```suggestion
    }

```

kevinthegreat1 reviewed (changes requested):
Thanks for your work so far. Once you address these it should be good to go!

UpFault commented:
jesus, was supposed to be 1 final commit, anyways, took the changes you suggested, added them and fixed the yPos issue I was having which is why it wasn't sending the player feedback or copying to the clipboard, everything should be good now.

UpFault reviewed (commented):
everything looks good
